src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1276,170 +1249,411 @@ end subroutine eigencalving
+               iEdge = edgesOnCell(iNeighbor, iCell)
+               if (li_mask_is_dynamic_ice(edgeMask(iEdge))) then
+                  edgeLengthScaling = scale_edge_length(angleEdge(iEdge), uReconstructX(1,iCell), uReconstructY(1,iCell))
+                  calvLengthEdge = edgeLengthScaling * dvEdge(iEdge)
+                  calvLengthCell = calvLengthCell + calvLengthEdge
+               endif
+            enddo
+
+            ! Now pass along the required calving volume relative to the interface length
+            do iNeighbor = 1, nEdgesOnCell(iCell)
+               iEdge = edgesOnCell(iNeighbor, iCell)
+               jCell = cellsOnCell(iNeighbor, iCell)
+               if (li_mask_is_dynamic_ice(edgeMask(iEdge))) then
+                  edgeLengthScaling = scale_edge_length(angleEdge(iEdge), uReconstructX(1,iCell), uReconstructY(1,iCell))
+                  calvLengthEdge = edgeLengthScaling * dvEdge(iEdge)
+                  if (requiredCalvingVolumeDynEdge(iEdge) > 0.0_RKIND) then
+                     call mpas_log_write("Unexpectedly found a dynamic edge that already has calving assigned to it." // &
+                           "  There is a flaw in apply_calving_velocity that needs to be fixed!", MPAS_LOG_ERR)
+                     err_tmp = 1
+                     err = ior(err, err_tmp)
                   endif
-               enddo
-               !TODO: need to recursively distribute across neighbors until fully depleted :(
-            endif
+                  requiredCalvingVolumeDynEdge(iEdge) = calvLengthEdge / calvLengthCell * volumeAvailableToPass
+                  uncalvedVolumeNonDynCell(iCell) = uncalvedVolumeNonDynCell(iCell) - requiredCalvingVolumeDynEdge(iEdge)
+                  !call mpas_log_write("   Passed calving $r from non-dynamic cell $i to dynamic cell $i", &
+                  !        realArgs=(/requiredCalvingVolumeDynEdge(iEdge)/), intArgs=(/iCell, jCell/))
+               endif
+            enddo
+         endif
+      enddo
 
-            ! If we didn't calve enough ice, record that to allow assessment of how bad that is.
-            if (volumeLeft > 0.0_RKIND) then
-               uncalvedVolume(iCell) = 0.0_RKIND
-            endif
+      ! c. Now apply calving to each cell
+      do iCell = 1, nCellsSolve
+         if ( (li_mask_is_floating_ice(cellMask(iCell)) .and. li_mask_is_dynamic_ice(cellMask(iCell)) .and. &
+                 li_mask_is_dynamic_margin(cellMask(iCell)) ) .or. (groundedMarineMarginMask(iCell) == 1) ) then
+            do iNeighbor = 1, nEdgesOnCell(iCell)
+               iEdge = edgesOnCell(iNeighbor, iCell)
+               ! No need to check what type of edge this is - only required edges are nonzero
+               requiredCalvingVolumeDynCell(iCell) = requiredCalvingVolumeDynCell(iCell) + &
+                          requiredCalvingVolumeDynEdge(iEdge) ! Keep running total
+            enddo
+            ! c. Apply calving here
```

Comment:
Same comment as above (line 1456). Need to be clear that we are updating accounting, but not actually removing the thickness yet.

Reply:
done